components/button/floating-buttons.js

```diff
@@ -0,0 +1,236 @@
+			 */
+			minHeight: {
+				reflect: true,
+				type: String,
+				attribute: 'min-height'
+			},
+
+			_viewportIsAtLeastMinHeight: {
+				reflect: true,
+				type: Boolean
+			}
+		};
+	}
+
+	static get styles() {
+		return [ labelStyles, buttonStyles,
+			css`
+			:host {
+				box-sizing: border-box;
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-floating-buttons-container {
+				border-top: 1px solid transparent;
+				display: block;
+				margin: 0 auto;
+				width: 100%;
+			}
+			.d2l-floating-buttons-container.d2l-floating-buttons-floating {
+				animation: d2l-floating-buttons-animation 500ms ease-out;
+				-webkit-animation: d2l-floating-buttons-animation 500ms ease-out;
+				background-color: #ffffff;
+				background-color: rgba(255, 255, 255, 0.88);
+				border-top-color: var(--d2l-color-mica);
+				bottom: 0;
+				box-shadow: 0 -2px 4px rgba(86, 90, 92, .2);
+				left: 0;
+				position: fixed;
+				right: 0;
+				z-index: 999;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			.d2l-floating-buttons-inner-container ::slotted(button),
+			.d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-right: var(--d2l-button-spacing) !important;
+				margin-bottom: var(--d2l-button-spacing) !important;
+			}
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-left: var(--d2l-button-spacing) !important;
+				margin-right: 0 !important;
+			}
+			@keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					transform: translate(0,0);
+				}
+			}
+			@-webkit-keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					-webkit-transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					-webkit-transform: translate(0,0);
+				}
+			}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this._container = null;
+		this._isRTL = false;
+		this._spacer = null;
+		this.minHeight = '500px';
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('resize', this._reposition);
+		window.removeEventListener('scroll', this._reposition);
+	}
+
+	render() {
+		return html`
+			<lit-media-query
```

Comment:
The `lit-media-query` wires up a resize event handler in its [firstUpdated](https://github.com/Victor-Bernabe/lit-media-query/blob/master/lit-media-query.js#L48), but it never removes the event listener if the `lit-media-query` is removed from the DOM, in such case it would be leaking memory.

Perhaps we can get by with a standard media-query in CSS?  I haven't compared `lit-media-query` to `iron-media-query` so I'm not sure if there is parity, but it looks like there's not much to it, so if we do indeed need something like this, I think we could create something better.  It may not even need to be an element... it could be a mixin or a simply little lib - after all, `lit-media-query` is a non-visible element with `display: none;`.  Being an element doesn't appear to serve it any purpose.